src/components/Wishlist-card/wishlist-card.jsx

```diff
@@ -0,0 +1,72 @@
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../contexts/auth-context";
+import { useWishlist } from "../../contexts/wishlist-context";
+import "../../pages/Product-listing/product-listing.css";
+import axios from "axios";
+
+const WishlistCard = ({ productcard }) => {
+  const { title, price, rating, productImg } = productcard;
+
+  const {
+    auth: { Authenticated, token },
+  } = useAuth();
+  const { dispatch: Wishlistdispatch } = useWishlist(); //renaming dispatch for better readibility
+  const navigate = useNavigate();
+
+  const deleteFromWishlist = async (id) => {
+    if (Authenticated) {
+      try {
+        const response = await axios.delete(`/api/user/wishlist/${id}`, {
+          headers: { authorization: token },
+        });
```

Comment:
`deleteFromWishlist` function is being called at two places. So you can place it in the context or make a util function for it.